### ISSUE TITLE
INVALID_APP_ID when running on 2nd device

### ISSUE THREAD

**Issue by stevenhewitt** [spotify/android-sdk#180]
I am working on an app that uses the Spotify SDK.  When I run it on the first android device I have, it works perfectly.  But when I tried running it on a 2nd, different android device, it returned INVALID_APP_ID error whenever I got the onActivityResult for authenticating.  Since it's being compiled from the same computer the android package and SHA1 fingerprint is the same; being a different phone shouldn't make a difference should it?

I've tried it with the same Spotify account, but the same error still occurs. When I run it with no account logged into spotify (on both devices), I instead get the error NOT_LOGGED_IN.

**Comment by kaaes** [comment]
@stevenhewitt yeas, it should work on any device as long as it's signed using the same certificate. Could you check if your app has the same signature on both devices?

**Comment by Inobtenio** [comment]
Hello. Same thing here but it happens when the device has Spotify App installed, everything works well with Webview login. This on every test device.

**Comment by stevenhewitt** [comment]
I investigated this more today, and it turns out on the My Applications page the Android Package and SHA1 fingerprint weren't getting saved (I must've mistaken that just pressing save was enough instead of needing to press add and then save).  However, the app still ran on my phone despite not having any fingerprint set; it only started giving me errors on a different device.  I then tested this more with a new test Spotify App, and was able to login and play songs with spotify using the new client_id/callback without having set a single fingerprint on both phones.

The original problem I fixed by just adding and saving the fingerprint, but there might be another problem with apps being sometimes validated without a fingerprint, which is what made me not realize my mistake sooner.
